com.unity.shadergraph/Documentation~/Refract-Node.md

```diff
@@ -0,0 +1,42 @@
+# Refract Node
+
+## Description
+
+Generates a refraction [Ibn Sahl-Snell-Descartes's Law](https://en.wikipedia.org/wiki/Snell%27s_law). Based on an **Incident** vector, the **Normal** of the surface and **Eta** we produce the refracted vector.
+
+Based on Ibn Sahl's Law, for a given refractive index of the medium, we can reach an angle where all the surface behave like a perfect mirror, that's why in **Safe** Mode generate a null vector, this implementation is here to not double the reflection intensity which is handle by Fresnel.
```

Comment:
This needs some work. If I'm understanding correctly... 

1. For any refractive index for a given medium, there is an angle where the surface behaves like a perfect mirror. 
2. There is a "Safe Mode" for this node that will return a null vector. 

I don't know what you mean by "this implementation is here to not double the reflection intensity which is handle by Fresnel"

Reply:
Safe mode is here to not have NaN and avoid the square root of a negative number.

"this implementation is here to not double the reflection intensity which is handle by Fresnel":
I could change the code to return the ray that implies the reflection (from total reflection), but I decided to not do that, because that behavior should be handle by the reflection from a better fresnel code.

Reply:
>The Refract node generates a refraction using an **Incident** vector, a normalized **Normal** vector of the surface, and the refractive index ratio, or **Eta** and produces a new refracted vector. The node uses the principles described in [Sahl-Snell's Law](https://en.wikipedia.org/wiki/Snell%27s_law). 

> Based on Sahl-Snell's Law, for a medium's given refractive index, there is an angle where the surface behaves like a perfect mirror. When you set the Refract node's Mode to **Safe**, the node generates a null vector when it reaches this critical angle. 

Please change the text here to the above.